File: ChipmeoPOS/src/lib/types/customer.ts
export interface Customer {
	id: number;
	fullName: string;
	phone?: string;
	email: string;
	avatarUrl?: string;
	points: number;
	isActive: boolean;
	createdAt?: string;
}

export interface CustomerRegisterDto {
	fullName: string;
	email: string;
	password: string;
	phone?: string;
}

export interface CustomerLoginDto {
	email: string;
	password: string;
}

export interface CustomerLoginResult {
	customer: Customer;
	token: string;
}

export interface CustomerUpdateDto {
	fullName?: string;
	phone?: string;
	avatarUrl?: string;
}

File: ChipmeoApis/ChipmeoApis.Core/Entities/Customer.cs
namespace ChipmeoApis.Core.Entities;

public partial class Customer
{
    public int Id { get; set; }

    public string FullName { get; set; } = null!;

    public string? Phone { get; set; }

    public string Email { get; set; } = null!;

    public string PasswordHash { get; set; } = null!;

    public string? AvatarUrl { get; set; }

    public int? Points { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}




